monkey/infection_monkey/network/ping_scanner.py

```diff
@@ -1,79 +1,83 @@
-
-        return 0 == subprocess.call(
-            ping_cmd,
-            stdout=self._devnull,
-            stderr=self._devnull,
-        )
-
-    def get_host_fingerprint(self, host):
-        ping_cmd = self._build_ping_command(host.ip_addr)
-        logger.debug(f"Running ping command: {' '.join(ping_cmd)}")
-
-        # If stdout is not connected to a terminal (i.e. redirected to a pipe or file), the result
-        # of os.device_encoding(1) will be None. Setting errors="backslashreplace" prevents a crash
-        # in this case. See #1175 and #1403 for more information.
-        encoding = os.device_encoding(1)
-        sub_proc = subprocess.Popen(
-            ping_cmd,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            text=True,
-            encoding=encoding,
-            errors="backslashreplace",
-        )
-
-        logger.debug(f"Retrieving ping command output using {encoding} encoding")
-        output = " ".join(sub_proc.communicate())
-        regex_result = self._ttl_regex.search(output)
-        if regex_result:
-            try:
-                ttl = int(regex_result.group(0))
-                if ttl <= LINUX_TTL:
-                    host.os["type"] = "linux"
-                else:  # as far we we know, could also be OSX/BSD but lets handle that when it
-                    # comes up.
-                    host.os["type"] = "windows"
-
-                host.icmp = True
-
-                return True
-            except Exception as exc:
-                logger.debug("Error parsing ping fingerprint: %s", exc)
-
-        return False
-
-    def _build_ping_command(self, ip_addr):
-        return ["ping", PING_COUNT_FLAG, "1", PING_TIMEOUT_FLAG, str(self._timeout), ip_addr]
+def ping(host: str, timeout: float) -> PingScanData:
+    if "win32" == sys.platform:
+        timeout = math.floor(timeout * 1000)
+
+    ping_command_output = _run_ping_command(host, timeout)
+
+    ping_scan_data = _process_ping_command_output(ping_command_output)
+    logger.debug(f"{host} - {ping_scan_data}")
+
+    return ping_scan_data
+
+
+def _run_ping_command(host: str, timeout: float) -> str:
+    ping_cmd = _build_ping_command(host, timeout)
+    logger.debug(f"Running ping command: {' '.join(ping_cmd)}")
+
+    # If stdout is not connected to a terminal (i.e. redirected to a pipe or file), the result
+    # of os.device_encoding(1) will be None. Setting errors="backslashreplace" prevents a crash
+    # in this case. See #1175 and #1403 for more information.
+    encoding = os.device_encoding(1)
+    sub_proc = subprocess.Popen(
+        ping_cmd,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        text=True,
+        encoding=encoding,
+        errors="backslashreplace",
+    )
+
+    logger.debug(f"Retrieving ping command output using {encoding} encoding")
+    # The underlying ping command should timeout within the specified timeout. Setting the timeout
```

Comment:
A good practice is to keep comments close to the code that's being commented. We read code in blocks and blocks are formed with empty lines. I would add a newline before this line and remove the one before try: